code/missionui/chatbox.cpp

```diff
@@ -402,28 +402,49 @@ size_t chatbox_get_msg_target_length(char *msg)
 
 }
 
+// send a msg to the game chat
+void chatbox_send_msg(char msg[150])
+{
+	// check if this message is supposed to have a recipient
+	int target = chatbox_get_msg_target_type(msg);
+	size_t target_length = chatbox_get_msg_target_length(msg);
+
+	// tack on the null terminator in the boundary case
+	size_t x = strlen(msg);
+	if (x >= CHATBOX_MAX_LEN) {
+		msg[CHATBOX_MAX_LEN - 1] = '\0';
+	}
+
+	// if I'm the server, then broadcast the packet
+	chatbox_recall_add(msg);
+	if (target != MULTI_MSG_EXPR) {
+		send_game_chat_packet(Net_player, msg, target);
+	} else {
+		// copy the name of the player the message is being sent to
+		char temp[150];
+		strncpy(temp, msg + 1, target_length - 2);
+		temp[target_length - 2] = '\0';
+		send_game_chat_packet(Net_player, msg, target, nullptr, temp);
+	}
+
+	chatbox_add_line(msg, MY_NET_PLAYER_NUM);
+}
+
 // automatically split up any input text, send it, and leave the remainder 
 void chatbox_autosplit_line()
 {
 	char msg[150];
```

Comment:
Going ahead and changing this to `char msg[CHATBOX_MAX_LEN]` would be a nice thing to do while your here. Looks like there is only the two places left to change the magic number.